### PR TITLE
Migrate from Tensorboard(X) to Weights & Biases

### PR DESCRIPTION
**Pull request by mshalm**
As the final task in Python 3.11 compatibility, we are migrating to logging via [Weights and Biases](https://docs.wandb.ai/) instead of Tensorboard due to compatibility issues of video logging. Weights and Biases has the added benefit of being cloud-based, allowing simultaneous viewing of all historical runs without using any local resources to run a server.

As part of this implementation, `tensorboard_manager.py` has been migrated to `wandb_manager.py`, and various variable names have been renamed in support of replacing `TensorboardManager` with `WeightsAndBiasesManager`. Finally, some changes to `Hyperparameter` were made to aid in logging experiment configurations automatically to aid reproducability.

### CODE REVIEW

File: dair_pll/wandb_manager.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        Args:
+            epoch: Current epoch in training process
+            scalars: Scalars to log.
+            videos: Videos to log.
+            meshes: Meshes to log.
+        """
+
+        _write_scalars(epoch, scalars)
+        _write_videos(epoch, videos)
+        _write_meshes(epoch, meshes)
```
Comment (ebianchi) [inline dair_pll/wandb_manager.py:83]: In the old `tensorboard_manager.py`, there was some checking to see if the provided `videos` and `meshes` in `TensorboardManager.update(epoch, scalars, videos, meshes)` were `not None`.  Does the `wandb.log()` function called in `_write_videos()` and `_write_meshes()` complain if they are given `None` types?

**Review by ebianchi — APPROVED** [review]
Clean code changes, and the W&B dashboard reports and provides everything as expected.

Aside from one small point in an inline comment which I expect to elicit a quick "yes", this looks good to me.

File: dair_pll/wandb_manager.py (hunk 1 above)
Comment (mshalm) [reply dair_pll/wandb_manager.py:83]: Unless I'm confused, this doesn't appear to be the case in `main` at the moment -- is there a previous commit where this was the case?

Comment (ebianchi) [reply dair_pll/wandb_manager.py:83]: Oops, you're totally right, this was something in my fork of the repo, as a consequence of my adding in an overlay plot for the loss components.  Then I agree this PR's implementation with W&B is just as robust as `main`'s implementation with Tensorboard.
